GRADLE_PLUGIN.md

```diff
@@ -1,10 +1,268 @@
+}
+```
+
+</div>
+
+The `kotlin-multiplatform` plugin also uses configurations under the hood but it also provides a `dependencies` block for each source set
+allowing configuring dependencies of this sources set:
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+kotlin.sourceSets {
+    commonMain {
+        dependencies {
+            implementation 'org.sample.test:mylibrary:1.0'
+        }
+    }
+    
+    commonTest {
+        dependencies {
+            implementation 'org.sample.test:testlibrary:1.0'
+        }
+    }
+}
+```
+
+</div>
+
+Note that a dependency declared for `commonMain` or `commonTest` source set must be published using the `kotlin-multiplatform` plugin.
+If you want to use libraries published by the `kotlin-platform-native` plugin, you need to declare a separate source set for common native sources.
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+kotlin.sourceSets {
+    // Create a common source set used by native targets only.
+    nativeMain {
+        dependsOn commonMain
+        dependencies {
+            // Depend on a library published by the kotlin-platform-naive plugin.
+            implementation 'org.sample.test:mylibrary:1.0'
+        }
+    }
+
+    // Configure all native platform sources sets to use it as a common one.
+    linuxX64Main.dependsOn nativeMain
+    macosX64Main.dependsOn nativeMain
+    //...
+}
+```
+
+</div>
+
+See more info about dependencies at the [corresponding page](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#adding-dependencies).
+
+### Output kinds
+
+With the `kotlin-platform-native` plugin output kinds are specified as a list in properties of a component:
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+components.main {
+    // Compile the component into an executable and a Kotlin/Native library.
+    outputKinds = [EXECUTABLE, KLIBRARY]
+}
+```
+
+</div> 
+
+With the `kotlin-multiplatform` plugin a compilation always produces a `*.klib` file. A separate `binaries` block is used to configure what
+final native binaries should be produced by each target. Each binary can be configured independently including linker options, executable entry point etc.
+
+<div class="sample" markdown="1" theme="idea" mode="groovy">
+
+```groovy
+kotlin {
+    macosX64 {
+        binaries {
+            executable {
+                // Binary configuration: linker options, name, etc.
+            }
+            framework {
+                // ...
+            }
+            
+        }
+    }
+}
+```
+
+</div> 
+
+See more about native binaries declaration at the [corresponding page](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#building-final-native-binaries).
+
+
+### Publishing
+
+Both `kotlin-platform-native` and `kotlin-multiplatform` plugins automatically set up artifact publication when the `maven-publish` plugin is applied.
```

Comment:
A link to [Publishing a Multiplatform Library](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#publishing-a-multiplatform-library) may be useful here.